### PR TITLE
[FIX] web: Reconcilication widget fail on astrophe as a thousands seperator

### PR DESCRIPTION
**Pull request by assellalou**
__Reproduction__:

- Setup new DB and change the Separator to anastrophe ```'```
- Browse to accounting -> Reconciliation (journal doesn't matter)
- Find an entry with a minimum of 4 figures to include a thousand separator
- Choose the entry and click on Manual Entry

__Result__: TB -> ```X``` is not a correct float

 This was a result of an unmatched behavior between the two functions ParseNumber and ParseMonetary which results in HTML special char (separator) not being removed.

opw-2725822

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by caburj — COMMENTED** [review]
I actually can't reproduce the issue in the latest runbot (Odoo 15) build.
- Joseph (jcb)

File: addons/web/static/tests/fields/parsers_tests.js [hunk 1]
```diff
@@ -117,6 +117,12 @@ QUnit.module("Fields", (hooks) => {
         assert.strictEqual(parseMonetary("$&nbsp;125.00", { currencyId: 3 }), 125);
         assert.strictEqual(parseMonetary("1,000.00&nbsp;€", { currencyId: 1 }), 1000);
 
+        // same tests with a different thousand separator
+        assert.strictEqual(parseMonetary("1'000.00"), 1000);
+        assert.strictEqual(parseMonetary("1'000'000.00"), 1000000);
+        assert.strictEqual(parseMonetary("$&nbsp;125'000.00", { currencyId: 3 }), 125);
```
[the remark this answers is not in the document]
Comment (caburj) [reply addons/web/static/tests/fields/parsers_tests.js:123]: One more thing. You should mock the "thousands separator", in your case, using the single quote symbol.

**Comment by assellalou** [comment]
> I actually can't reproduce the issue in the latest runbot (Odoo 15) build.
> 
> * Joseph (jcb)

it was solved by just not using the custom separator in the widget I guess as I couldn't find the same separator I set.

**Comment by assellalou** [comment]
this could be closed.

**Pull request closed without merging**